### PR TITLE
[4] Non-Controversial CodeStyle Cleanup for /components

### PR DESCRIPTION
**Pull request by PhilETaylor**
[4] Non-Controversial CodeStyle Cleanup for /components

Code review

### CODE REVIEW

File: components/com_contact/src/View/Contact/HtmlView.php [hunk 1]
```diff
@@ -170,15 +170,15 @@ public function display($tpl = null)
...
+		if (\count($errors = $this->get('Errors')))
 		{
 			throw new GenericDataException(implode("\n", $errors), 500);
 		}
 
 		// Check if access is not public
 		$groups = $user->getAuthorisedViewLevels();
 
-		if ((!in_array($item->access, $groups)) || (!in_array($item->category_access, $groups)))
+		if ((!\in_array($item->access, $groups)) || (!\in_array($item->category_access, $groups)))
```
Comment (Quy) [inline components/com_contact/src/View/Contact/HtmlView.php]:
```suggestion
		if (!\in_array($item->access, $groups) || !\in_array($item->category_access, $groups))
```

File: components/com_contact/src/View/Contact/VcfView.php [hunk 2]
```diff
@@ -69,7 +69,7 @@ public function display($tpl = null)
 				$namearray = explode(' ', $name_and_midname);
 
 				$firstname = $namearray[0];
-				$middlename = (count($namearray) > 1) ? $namearray[1] : '';
+				$middlename = (\count($namearray) > 1) ? $namearray[1] : '';
```
Comment (Quy) [inline components/com_contact/src/View/Contact/VcfView.php]:
```suggestion
				$middlename = \count($namearray) > 1 ? $namearray[1] : '';
```

File: components/com_contact/src/View/Contact/VcfView.php [hunk 3]
```diff
@@ -78,9 +78,9 @@ public function display($tpl = null)
 		{
 			$namearray = explode(' ', $item->name);
 
-			$middlename = (count($namearray) > 2) ? $namearray[1] : '';
+			$middlename = (\count($namearray) > 2) ? $namearray[1] : '';
```
Comment (Quy) [inline components/com_contact/src/View/Contact/VcfView.php]:
```suggestion
			$middlename = \count($namearray) > 2 ? $namearray[1] : '';
```

File: components/com_content/src/Model/ArticlesModel.php [hunk 4]
```diff
@@ -456,7 +456,7 @@ protected function getListQuery()
 				$query->bind(':categoryId', $categoryId, ParameterType::INTEGER);
 			}
 		}
-		elseif (is_array($categoryId) && (count($categoryId) > 0))
+		elseif (\is_array($categoryId) && (\count($categoryId) > 0))
```
Comment (Quy) [inline components/com_content/src/Model/ArticlesModel.php]:
```suggestion
		elseif (\is_array($categoryId) && \count($categoryId) > 0)
```

File: components/com_content/src/Model/ArticlesModel.php [hunk 5]
```diff
@@ -582,7 +582,7 @@ protected function getListQuery()
 		}
 
 		// Process the filter for list views with user-entered filters
-		if (is_object($params) && ($params->get('filter_field') !== 'hide') && ($filter = $this->getState('list.filter')))
+		if (\is_object($params) && ($params->get('filter_field') !== 'hide') && ($filter = $this->getState('list.filter')))
```
Comment (Quy) [inline components/com_content/src/Model/ArticlesModel.php]:
```suggestion
		if (\is_object($params) && $params->get('filter_field') !== 'hide' && $filter = $this->getState('list.filter'))
```

File: components/com_content/tmpl/article/default_links.php [hunk 6]
```diff
@@ -24,15 +24,15 @@
...
+			array($urls->urlc, $urls->urlctext, $urls->targetc, 'c'),
 			);
 			foreach ($urlarray as $url) :
 				$link = $url[0];
 				$label = $url[1];
 				$target = $url[2];
 				$id = $url[3];
 
-				if ( ! $link) :
+				if (! $link) :
```
Comment (Quy) [inline components/com_content/tmpl/article/default_links.php]:
```suggestion
				if (!$link) :
```

File: components/com_newsfeeds/src/View/Newsfeed/HtmlView.php [hunk 7]
```diff
@@ -164,7 +164,7 @@ public function display($tpl = null)
 		// Check the access to the newsfeed
 		$levels = $user->getAuthorisedViewLevels();
 
-		if (!in_array($item->access, $levels) || (in_array($item->access, $levels) && (!in_array($item->category_access, $levels))))
+		if (!\in_array($item->access, $levels) || (\in_array($item->access, $levels) && (!\in_array($item->category_access, $levels))))
```
Comment (Quy) [inline components/com_newsfeeds/src/View/Newsfeed/HtmlView.php]:
```suggestion
		if (!\in_array($item->access, $levels) || (\in_array($item->access, $levels) && !\in_array($item->category_access, $levels)))
```

File: components/com_users/src/Model/ProfileModel.php [hunk 8]
```diff
@@ -128,7 +128,7 @@ public function getForm($data = array(), $loadData = true)
 
 		if ($username)
 		{
-			$isUsernameCompliant  = !(preg_match('#[<>"\'%;()&\\\\]|\\.\\./#', $username) || strlen(utf8_decode($username)) < 2
+			$isUsernameCompliant  = !(preg_match('#[<>"\'%;()&\\\\]|\\.\\./#', $username) || \strlen(utf8_decode($username)) < 2
```
Comment (Quy) [inline components/com_users/src/Model/ProfileModel.php]:
```suggestion
			$isUsernameCompliant = !(preg_match('#[<>"\'%;()&\\\\]|\\.\\./#', $username) || \strlen(utf8_decode($username)) < 2
```

File: components/com_banners/src/Model/BannersModel.php [hunk 9]
```diff
@@ -171,7 +171,7 @@ protected function getListQuery()
 					->bind(':categoryId', $categoryId, ParameterType::INTEGER);
 			}
 		}
-		elseif (is_array($categoryId) && (count($categoryId) > 0))
+		elseif (\is_array($categoryId) && (\count($categoryId) > 0))
```
Comment (Quy) [inline components/com_banners/src/Model/BannersModel.php]:
```suggestion
		elseif (\is_array($categoryId) && \count($categoryId) > 0)
```

**Comment by joomdonation** [comment]
@PhilETaylor Do you think PR like this will cause merge conflicts with:

- Existing PRs
- Or when @wilsonge merge the changes from staging and 3.10 to 4.0-dev

For these reasons, maybe we should do it a bit later ?

**Comment by PhilETaylor** [comment]
> For these reasons, maybe we should do it a bit later ?

That was @wilsonge excuse was almost 5 years ago when I tried to improve the quality of the code by importing classes. 

Im sorry but I dont believe it should be left. Merge conflicts are easy - very easy - to resolve, and this PR fully complies with all the Joomla Coding Standards and best practice and there is no reason why it should not be tested and merged. 

Ultimately its not my decision, however if this PR and the others are rejected, it says a lot about the quality of the code and the project as a whole.

**Comment by rdeutz** [comment]
I don't think it is the right moment for massive code (style) changes, we should concentrate on the important things to get 4.0.0 ready. Changing a lot of files will make upstream merges more complicated as they are already. I don't will mege any of the code style PRs. We can revisit this after we have released 4.0.0

**Comment by PhilETaylor** [comment]
Well that's your opinion, and as I said - speaks volumes about the project. 

These changes are bringing the code in line with the Joomla Coding standards. The code should already be complying with those standards. 

Ive been hearing the same excuses for over 5 years now. There will never be a right time.

**Comment by PhilETaylor** [comment]
> We can revisit this after we have released 4.0.0

That will never happen.

**Comment by PhilETaylor** [comment]
> I don't think it is the right moment for massive code (style) changes

So you are happy that Joomla code doesn't comply with Joomla Coding Standards... Got it.

**Comment by PhilETaylor** [comment]
Closing as no one has the balls to merge this.

**Comment by PhilETaylor** [comment]
> we should concentrate on the important things to get 4.0.0 ready

Like the 1980s FTP Layer??? Yeah right...

**Comment by rdeutz** [comment]
> So you are happy that Joomla code doesn't comply with Joomla Coding Standards... Got it.

Would be nicer, if you don't put words into my mouth.

**Comment by PhilETaylor** [comment]
It would be nicer if maintainers maintained, and merged, but here we are...

**Comment by PhilETaylor** [comment]
Im struggling to think of another project that blocks progress "because it would make merging harder"... 

Merging = Selecting from the left or right and clicking save. 

Merging is not difficult !

**Pull request closed without merging**